alerting/src/main/kotlin/org/opensearch/alerting/model/action/Action.kt

```diff
@@ -171,7 +179,8 @@ data class Action(
                 requireNotNull(messageTemplate) { "Action message template is null" },
                 throttleEnabled,
                 throttle,
-                id = requireNotNull(id)
+                id = requireNotNull(id),
+                actionExecutionPolicy = requireNotNull(actionExecutionPolicy) { "Action execution policy is null" }
```

Comment:
What happen when for old Actions which doesn't have the actionExecutionPolicy? Is this change backward compatible?

Reply:
So Query-Level Monitors don't actually use the `ActionExecutionPolicy` as of now. If we make the change in the future to allow it, we'd update the default based on Monitor type to be compatible with both at that time.

Reply:
To be on a safer side, we should try to keep the behavior for existing query-level monitors as it is i.e. to not add any default field for `actionExecutionPolicy` in the data class for query-level monitors. This will ensure both forward and backward compatibility issues.

Reply:
Yeah, that makes sense. I'll merge in this change for now since it doesn't reflect the full usage of `ActionExecutionPolicy`. Once we get to the PR with the changes to `MonitorRunner`, I think we can update `actionExecutionPolicy` to be nullable and set the default configuration at runtime if it is not set.